src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -185,7 +186,16 @@ public void OnStartActivity(Activity activity, object payload)
             // see the spec https://github.com/open-telemetry/semantic-conventions/blob/v1.21.0/docs/http/http-spans.md
             if (this.emitNewAttributes)
             {
-                activity.SetTag(SemanticConventions.AttributeHttpRequestMethod, HttpTagHelper.GetNameForHttpMethod(request.Method));
+                if (RequestMethodHelper.TryResolveHttpMethod(request.Method.Method, out var httpMethod))
```

Comment:
This code is very confusing to read. It is a "Try" method which returns true or false however both branches use the result as the tag. You have to go and read the code to understand that a false return actually means "_OTHER" was returned.

How could we improve this with readability/maintainability in mind?

Reply:
Same. At first glance I was expecting it to be `null` and had to check the implementation.

Reply:
fixed.

Reply:
I think what you did is better, but now it is violating DRY and causing the "_OTHER" logic to be leaked/duplicated in multiple spots.

Why not something like...

```csharp
public (string resolvedMethod, bool isKnown) ResolveHttpMethod(string originalMethod) {}
```

Used like this...

```csharp
                (string resolvedMethod, bool isKnown) = RequestMethodHelper.ResolveHttpMethod(request.Method);

                activity.SetTag(SemanticConventions.AttributeHttpRequestMethod, resolvedMethod);
                if (!isKnown)
                {
                    activity.SetTag(SemanticConventions.AttributeHttpRequestMethodOriginal, request.Method);
                }
```

Reply:
Good point. I will update it.